sdk/core/azure-core/test/ut/md5.cpp

```diff
@@ -118,3 +120,32 @@ TEST(Md5Hash, CtorDtor)
     Md5Hash instance;
   }
 }
+
+TEST(Md5Hash, multiThread)
+{
+  auto hashThreadRoutine = [](int sleepFor) {
```

Comment:
Any bug fix in this PR that would be good to call out in the CL?

Reply:
there's no bug fix.
The previous impl was thread safe as well.